reseq_ros2/launch/reseq_launch.py

```diff
@@ -1,24 +1,80 @@
 from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
+import sys
+import yaml
+from yaml import SafeLoader
 from launch_ros.actions import Node
 from ament_index_python.packages import get_package_share_directory
 
+share_folder = get_package_share_directory("reseq_ros2")
+
+def get_config():
+    args = sys.argv
+    if len(args)>1:
+        filename = share_folder + args[1]
+    else:
+        filename = share_folder + "/config/reseq_mk1_can.yaml"
+    return filename
+
+def parse_config(filename):
+    with open(filename) as f:
+        return yaml.load(f, Loader=SafeLoader)
+    
+def declare_parameters(config):
+    #Declare each parameter present in the yaml fle as a Launch Argument
+    args = []
+    for k,v in config.items():
+        if k=='geometric_consts':
+            for name, val in v.items():
+                args.append(DeclareLaunchArgument(name, default_value = str(val)))
+        elif k=='enea_consts':
+            for name, val in v.items():
+                args.append(DeclareLaunchArgument(name, default_value = str(val)))
+        elif k=='modules':
+            addresses = []
+            joints = []
+            endEffectors = []
+            for mod in v:
+                addresses.append(mod['address'])
+                if mod['hasJoint']:
+                    joints.append(mod['address'])
+                if mod['hasEndEffector']:
+                    endEffectors.append(mod['address'])
+            args.append(DeclareLaunchArgument('modules', default_value = str(addresses)))
+            args.append(DeclareLaunchArgument('joints', default_value = str(joints)))
+            args.append(DeclareLaunchArgument('end_effectors', default_value = str(endEffectors)))
+        elif k=='canbus':
+            args.append(DeclareLaunchArgument('can_channel', default_value = v['channel']))
+    return args
```

Comment:
I think that it would be better to have one single parameter: the config filename, because different configurations have different parameters and we will never use the launch file providing all the parameters

Reply:
Okay, so I will pass only the name of config file (declaring one singular launch argument) and I’ll restore the parsing of config file in each node.

Reply:
Well I think that passing the argument is correct and that the config file should be parsed in the launch file as you correctly did

The thing which I do not think we should do is declaring the parsed constants as launch parameters since we will never use those. The only launch parameter should be the config file name

Reply:
Okay,now I think I understand; I'll maintain the parameters passed to the nodes as I did, but not declaring them as launch arguments. Moreover I'll pass also the config name as parameter, declaring it as launch argument.

Reply:
I do not think we should pass it as a parameter if we already pass the other values. Let's just declare it. About the rest, perfect!